FreeRTOS/FreeRTOS-Plus-TCP
# Update STM32Hxx port to use HW MAC hashtable filter

Description
-----------
This PR updates the STM32Hxx port to use hardware MAC hashtable filter to allow list MAC addresses, instead of using the perfect filter which has limited MAC address registers.

References:
[RM0468 STM32H723/733, STM32H725/735 and STM32H730 Value line advanced Arm®-based 32-bit MCUs](https://www.st.com/resource/en/reference_manual/rm0468-stm32h723733-stm32h725735-and-stm32h730-value-line-advanced-armbased-32bit-mcus-stmicroelectronics.pdf)

Test Steps
-----------
Tested with demo project on STM32H743Zi with IPv6/IPv4 network. (RA/ICMPv6)

Checklist:
----------


- ~~[ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.~~ (N/A)

Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
